x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -116,23 +114,23 @@ public String toString() {
     public static final String SQL_TABLE = "TABLE";
     public static final String SQL_VIEW = "VIEW";
 
-    private static final IndicesOptions INDICES_ONLY_OPTIONS = new IndicesOptions(
-        EnumSet.of(Option.ALLOW_NO_INDICES, Option.IGNORE_UNAVAILABLE, Option.IGNORE_ALIASES, Option.IGNORE_THROTTLED),
-        EnumSet.of(WildcardStates.OPEN)
-    );
-    private static final IndicesOptions FROZEN_INDICES_OPTIONS = new IndicesOptions(
-        EnumSet.of(Option.ALLOW_NO_INDICES, Option.IGNORE_UNAVAILABLE, Option.IGNORE_ALIASES),
-        EnumSet.of(WildcardStates.OPEN)
-    );
-
-    public static final IndicesOptions FIELD_CAPS_INDICES_OPTIONS = new IndicesOptions(
-        EnumSet.of(Option.ALLOW_NO_INDICES, Option.IGNORE_UNAVAILABLE, Option.IGNORE_THROTTLED),
-        EnumSet.of(WildcardStates.OPEN)
-    );
-    public static final IndicesOptions FIELD_CAPS_FROZEN_INDICES_OPTIONS = new IndicesOptions(
-        EnumSet.of(Option.ALLOW_NO_INDICES, Option.IGNORE_UNAVAILABLE),
-        EnumSet.of(WildcardStates.OPEN)
-    );
+    private static final IndicesOptions INDICES_ONLY_OPTIONS = IndicesOptions.newBuilder()
+        .concreteTargetOptions(IndicesOptions.ConcreteTargetOptions.ALLOW_UNAVAILABLE_TARGETS)
+        .wildcardOptions(IndicesOptions.WildcardOptions.newBuilder().resolveAliases(false))
+        .generalOptions(IndicesOptions.GeneralOptions.newBuilder().removeThrottled(true))
+        .build();
```

Comment:
I notice here that we do not specify that we allow no indices. I imagine that's because it defaults to true, but I think it might be worth it to go through all of these constants and set the properties explicitly, even if they match the defaults. If we ever wanted to change the defaults for whatever reason, it's much harder to keep track of which things are relying on them and which things would be negatively affected by the change.